controller/scheduler/scheduler.go

```diff
@@ -1359,6 +1447,9 @@ func (s *Scheduler) handleJobStatus(job *Job, status host.JobStatus) {
 			formation = s.handleFormation(ef)
 		}
 		job.Formation = formation
+		if proc, ok := formation.Release.Processes[job.Type]; ok {
+			job.Service = proc.Service
```

Comment:
I think this needs to be more complicated, we should only set this if there is a corresponding route. Also, if the route is later removed the scheduler will wait indefinitely for the backend to drain. I guess it comes down to needing to know if the backend is currently tracked by the router.